src/metabase/api/card.clj

```diff
@@ -994,11 +1000,15 @@ saved later when it is ready."
         schema-name       (get-setting-or-throw! :uploads-schema-name)
         filename-prefix   (or (second (re-matches #"(.*)\.csv$" filename))
                               filename)
-        table-name-prefix (str (get-setting-or-throw! :uploads-table-prefix) filename-prefix)
+        table-name        (-> (str (get-setting-or-throw! :uploads-table-prefix) filename-prefix)
+                              csv/unique-table-name)
+        schema+table-name (if (str/blank? schema-name)
+                            (str schema-name "." table-name)
+                            table-name)
         driver            (driver.u/database->driver database)
-        _                 (or (driver/database-supports? driver :csv-uploads nil)
+        _                 (or (driver/database-supports? driver :uploads nil)
                               (throw (Exception. (tru "Uploads are not supported on {0} databases." (str/capitalize (name driver))))))
-        table-name        (csv/load-from-csv driver db-id schema-name csv-file table-name-prefix)
+        _                 (csv/load-from-csv driver db-id schema+table-name csv-file)
```

Comment:
Refactor: previously csv/load-from-csv did the calculation of the table-name, and `csv/load-from-csv*` did the rest. But we don't need the extra layer of indirection so I removed csv/load-from-csv, and now we do the calculation of the table-name here in `upload-csv!`

Reply:
:+1: 

Soon™ we're going to have to deal with temp files getting crazy names, but that's not something to worry about in this PR.